File: ios/ChargeTracker/ChargeTrackerEventEmitter.m
//
//  ChargeTrackerEventEmitter.m
//  LiveActivityPOC
//
//  Created by Varun Kukade on 27/11/24.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ChargeTrackerEventEmitter, RCTEventEmitter)

+ (bool)requiresMainQueueSetup {
  return NO;
}

RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/ChargeTracker/ChargeTrackerEventEmitter.swift
//
//  ChargeTrackerEventEmitter.swift
//  LiveActivityPOC
//
//  Created by Varun Kukade on 27/11/24.
//

import Foundation

@objc(ChargeTrackerEventEmitter)
class ChargeTrackerEventEmitter: RCTEventEmitter {

  public static var emitter: ChargeTrackerEventEmitter?
  
  public static var events = ["onStopChargeInitiated"];

  override init() {
    super.init()
    ChargeTrackerEventEmitter.emitter = self
  }

  override func supportedEvents() -> [String]! {
    return ChargeTrackerEventEmitter.events;
  }
}
